chainercv/links/model/resnet/resnet.py

```diff
@@ -0,0 +1,281 @@
+from __future__ import division
+
+import numpy as np
+
+import chainer
+import chainer.functions as F
+from chainer import initializers
+import chainer.links as L
+
+from chainercv.links import Conv2DBNActiv
+from chainercv.links.model.resnet.resblock import ResBlock
+from chainercv.links import PickableSequentialChain
+from chainercv.utils import download_model
+
+
+# RGB order
+# This is channel wise mean of mean image distributed at
+# https://github.com/KaimingHe/deep-residual-networks
+_imagenet_mean = np.array(
+    [123.15163084, 115.90288257, 103.0626238],
+    dtype=np.float32)[:, np.newaxis, np.newaxis]
+
+
+class ResNet(PickableSequentialChain):
+
+    """Base class for ResNet Network.
+
+    This is a feature extraction link.
+    The network can choose output layers from set of all
+    intermediate layers.
+    The attribute :obj:`pick` is the names of the layers that are going
+    to be picked by :meth:`__call__`.
+    The attribute :obj:`layer_names` is the names of all layers
+    that can be picked.
+
+    Examples:
+
+        >>> model = ResNet50()
+        # By default, __call__ returns a probability score (after Softmax).
+        >>> prob = model(imgs)
+        >>> model.pick = 'res5'
+        # This is layer res5
+        >>> res5 = model(imgs)
+        >>> model.pick = ['res5', 'fc6']
+        >>> # These are layers res5 and fc6.
+        >>> res5, fc6 = model(imgs)
+
+    .. seealso::
+        :class:`chainercv.links.model.PickableSequentialChain`
+
+    When :obj:`pretrained_model` is the path of a pre-trained chainer model
+    serialized as a :obj:`.npz` file in the constructor, this chain model
+    automatically initializes all the parameters with it.
+    When a string in the prespecified set is provided, a pretrained model is
+    loaded from weights distributed on the Internet.
+    The list of pretrained models supported are as follows:
+
+    * :obj:`imagenet`: Loads weights trained with ImageNet and distributed \
+        at `Model Zoo \
+        <https://github.com/BVLC/caffe/wiki/Model-Zoo>`_.
+        This is only supported when :obj:`arch=='he'`.
+
+    Args:
+        model_name (str): Name of the resnet model to instantiate.
```

Comment:
How about `n_layer`? It takes one of `50, 101, 152` as integer. `model_name` is difficult to understand.